ewatercycle/models/abstract.py

```diff
@@ -68,6 +91,52 @@ def set_value(self, name: str, value: np.ndarray) -> None:
         """
         self.bmi.set_value(name, value)
 
+    def set_value_at_indices(self, name: str, indices: Iterable[int], value: np.ndarray) -> None:
+        """Specify a new value for a model variable at particular indices.
+
+        Args:
+            name: Name of variable
+            indices: The indices into the variable array
+            value: The new value for the specified variable.
+
+        """
+        indices = np.array(indices)
+        self.bmi.set_value_at_indices(name, indices, value)
+
+    def set_value_at_coords(self, name: str, lat: Iterable[float], lon: Iterable[float], values: np.ndarray) -> None:
+        """Specify a new value for a model variable at at lat/lon coordinates.
+
+        Args:
+            name: Name of variable
+            lat: Latitudinal value
+            lon: Longitudinal value
+            value: The new value for the specified variable.
+
+        """
+        indices = self.coords_to_indices(name, lat, lon)
+        self.set_value_at_indices(name, indices, values)
+
+    def conversion_feedback(self, coord_user, coord_converted):
+        print(f"Your coordinates {coord_user} match these model coordinates {coord_converted}.")
+
+    @abstractmethod
+    def coords_to_indices(self, name: str, lat: Iterable[float], lon: Iterable[float]) -> Iterable[int]:
+        """Converts lat/lon values to index.
+
+        Args:
+            lat: Latitudinal value
+            lon: Longitudinal value
+
+        """
+
```

Comment:
```suggestion
        """
        return NotImplementedError("nice message")

```
and remove the `@abstractmethod`, as I think it's okay for models to not implement this method.

Reply:
your suggestion is applied.